x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/cluster/routing/allocation/DataTierIT.java

```diff
@@ -0,0 +1,219 @@
+            .setResizeType(ResizeType.SHRINK)
+            .setSettings(Settings.builder()
+                .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, 1)
+                .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 0)
+                .build()).get();
+
+        ensureGreen(index + "-shrunk");
+
+        Settings idxSettings = client().admin().indices().prepareGetIndex().addIndices(index + "-shrunk")
+            .get().getSettings().get(index + "-shrunk");
+        // It should inherit the setting of its originator
+        assertThat(DataTierAllocationDecider.INDEX_ROUTING_INCLUDE_SETTING.get(idxSettings), equalTo(DataTier.DATA_WARM));
+
+        // Required or else the test cleanup fails because it can't delete the indices
+        client().admin().indices().prepareUpdateSettings(index, index + "-shrunk")
+            .setSettings(Settings.builder()
+                .put("index.blocks.read_only", false))
+            .get();
+    }
+
+    public void testTemplateOverridesDefaults() {
+        startWarmNode();
+
+        Template t = new Template(Settings.builder()
+            .put(DataTierAllocationDecider.INDEX_ROUTING_REQUIRE, DataTier.DATA_WARM)
+            .build(), null, null);
+        ComposableIndexTemplate ct = new ComposableIndexTemplate(Collections.singletonList(index), t, null, null, null, null, null);
+        client().execute(PutComposableIndexTemplateAction.INSTANCE,
+            new PutComposableIndexTemplateAction.Request("template").indexTemplate(ct)).actionGet();
+
+        client().admin().indices().prepareCreate(index).setWaitForActiveShards(0).get();
+
+        Settings idxSettings = client().admin().indices().prepareGetIndex().addIndices(index).get().getSettings().get(index);
+        assertThat(idxSettings.keySet().contains(DataTierAllocationDecider.INDEX_ROUTING_INCLUDE), equalTo(false));
+
+        // index should be yellow
+        ensureYellow(index);
+
+        client().admin().indices().prepareDelete(index).get();
+
+        t = new Template(Settings.builder()
+            .putNull(DataTierAllocationDecider.INDEX_ROUTING_INCLUDE)
+            .build(), null, null);
+        ct = new ComposableIndexTemplate(Collections.singletonList(index), t, null, null, null, null, null);
+        client().execute(PutComposableIndexTemplateAction.INSTANCE,
+            new PutComposableIndexTemplateAction.Request("template").indexTemplate(ct)).actionGet();
+
+        client().admin().indices().prepareCreate(index).setWaitForActiveShards(0).get();
+
+        idxSettings = client().admin().indices().prepareGetIndex().addIndices(index).get().getSettings().get(index);
+        assertThat(idxSettings.keySet().contains(DataTierAllocationDecider.INDEX_ROUTING_INCLUDE), equalTo(false));
+
+        ensureYellow(index);
+    }
+
+    public void startHotNode() {
+        Settings nodeSettings = Settings.builder()
+            .putList("node.roles", Arrays.asList("master", "data_hot", "ingest"))
```

Comment:
maybe switch between "data" and "data_hot" randomly here ... IIUC they are synonymous in this context ?

Reply:
They are synonymous in that they can both have hot data, but not for tests that may expect starting a hot node to mean there are no warm/cold/frozen nodes in the cluster. I'll rename these to `startHotOnlyNode` (for all the node types)